### PR TITLE
2) passed all task

### PR DESCRIPTION
**Pull request by AntonPanchishko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -9,8 +9,70 @@
...
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+
 </project>
```
Comment (mfedenko09) [inline pom.xml]: add empty line please 
```suggestion
</project>
```

File: src/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+language: java
+jdk:
+  - openjdk11
+script:
+  - mvn clean install
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (mfedenko09) [inline src/.travis.yml]:
```suggestion
    - $HOME/.m2
```
here same situation with empty line

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert movie entity", e);
```
Comment (mfedenko09) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]:
```suggestion
            throw new RuntimeException("Can't insert movie entity", e);
```
here we should throw our custom exception

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    public List<Movie> getAll() {
+
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> getAllMoviesQuery = session.createQuery("FROM Movie", Movie.class);
+            return getAllMoviesQuery.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all movies from db", e);
```
Comment (mfedenko09) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: same here

File: src/main/java/com/dev/cinema/model/Movie.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
+package com.dev.cinema.model;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Movie {
```
Comment (mfedenko09) [inline src/main/java/com/dev/cinema/model/Movie.java:11]:
```suggestion
public class Movie {
```
add here @ Table(name = "movie")
it's necessary  to name our table with lower case

File: src/main/java/com/dev/cinema/util/HibernateUtil.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    private static SessionFactory sessionFactory = initSessionFactory();
+
+    public HibernateUtil() {
+    }
+
+    private static SessionFactory initSessionFactory() {
+        try {
+            return new Configuration().configure().buildSessionFactory();
+        } catch (Exception e) {
+            throw new RuntimeException("Cant creating session factory", e);
```
Comment (mfedenko09) [inline src/main/java/com/dev/cinema/util/HibernateUtil.java:16]:
```suggestion
            throw new RuntimeException("Cant creating session factory", e);
```
here the same situation with custom exception

File: src/main/resources/hibernate.cfg.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        <property name = "hibernate.connection.url">jdbc:mysql://localhost/cinema?serverTimezone=UTC</property>
+        <property name = "hibernate.connection.username">root</property>
+        <property name = "hibernate.connection.password">1234</property>
+        <property name="show_sql">true</property>
+        <property name="hbm2ddl.auto">create-drop</property>
+
+        <mapping class="com.dev.cinema.model.Movie"/>
+
+    </session-factory>
+</hibernate-configuration>
```
Comment (mfedenko09) [inline src/main/resources/hibernate.cfg.xml]:
```suggestion
</hibernate-configuration>
```
add empty line please

**Review by mfedenko09 — CHANGES_REQUESTED** [review]

**Review by mfedenko09 — APPROVED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/homework/hibernate_01.html

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataBindingException("Cant insert entity in db", e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]:
```suggestion
            throw new DataBindingException("Cant insert entity in db " + movie, e);
```

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        } finally {
+            session.close();
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: let's add checking for null here

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    }
+
+    @Override
+    public List<Movie> getAll() {
+
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: remove the empty line at the beginning of the method

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/util/HibernateUtil.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    private static SessionFactory sessionFactory = initSessionFactory();
+
+    public HibernateUtil() {
+    }
+
+    private static SessionFactory initSessionFactory() {
+        try {
+            return new Configuration().configure().buildSessionFactory();
+        } catch (Exception e) {
+            throw new DataBindingException("Cant creating session factory", e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/util/HibernateUtil.java]: it isn't data problem, so let's throw RuntimeException here

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**